### PR TITLE
[FRONT] #463 fix room editing

### PR DESCRIPTION
**Pull request by ntchou**
- fix undefined room id
- refactor `Api`

### CODE REVIEW

**Review by HugoGresse — APPROVED** [review]

File: src/pages/AdminDashboard/roomsActions.ts [hunk 1]
```diff
@@ -20,22 +21,20 @@ export const getRooms = (t) => async (dispatch, getState) => {
 
     try {
         const rooms = await api.getRooms()
+        console.log('Rooms: ', rooms)
```
Comment (HugoGresse) [inline src/pages/AdminDashboard/roomsActions.ts:24]: remove?

File: src/services/api.ts [hunk 2]
```diff
@@ -2,45 +2,34 @@ import { JWTToken } from '../../types/JWT'
...
-        return this.post('/rooms/new', {
-            name: name,
+    async createRoom(room: Room): Promise<NewRoomResponse> {
+        const { name, hostName, destinations } = room
+        return this.apiClient.post('/rooms/new', {
+            name,
             hostName,
             destinations: JSON.stringify(destinations),
-            password,
+            password: 'test-password', //TODO set a password ?
```
Comment (HugoGresse) [inline src/services/api.ts:22]: no password in the UI for the moment

**Review by HugoGresse — CHANGES_REQUESTED** [review]

File: src/pages/AdminDashboard/roomsActions.ts (hunk 1 above)
Comment (ntchou) [reply src/pages/AdminDashboard/roomsActions.ts:24]: merge from matthia branch

File: src/services/api.ts (hunk 2 above)
Comment (ntchou) [reply src/services/api.ts:22]: merge from matthia branch. I'll fix it

**Review by MattiaPrimavera — APPROVED** [review]

File: src/services/api.ts (hunk 2 above)
Comment (MattiaPrimavera) [reply src/services/api.ts]: yep we just discussed about that :)

**Pull request merged**